### PR TITLE
Enhancement/add more components

### LINKED ISSUE #5: Add More Components
DevUI currently has a limited set of components. We need **more shadcn/ui components** like:

* Tabs
* Dialog / Modal
* Dropdowns / Select
* Accordion
* Toast / Notifications

**Task:**

* Create the component
* Add live preview
* Add syntax-highlighted, copyable code snippet
* Ensure responsiveness and dark theme support

### PR DESCRIPTION
**Pull request by krishkaneki**
## Description
Added more shadcn/ui components to DevUI including Tabs, Dropdown Menu, and Popover components. Each component includes live interactive previews, syntax-highlighted code snippets, comprehensive prop documentation, and full support for dark theme and responsive design.

## Related Issue
Fixes #5

## Changes Made
- [x] Added **Tabs** component with account/password example and live preview
- [x] Added **Dropdown Menu** component with checkbox and radio group support
- [x] Added **Popover** component with form dimensions example
- [x] Installed required Radix UI dependencies (`@radix-ui/react-dropdown-menu`, `@radix-ui/react-popover`)
- [x] Updated `components.tsx` data file with new component entries
- [x] Fixed duplicate typography component issue
- [x] Fixed syntax errors in components array

### New Components Preview
<div align="center">

**Tabs Component**
Tabs Component Preview

**Dropdown Menu & Popover Components**
Dropdown Menu and Popover Components Preview

</div>

## Checklist

- [x] Only the necessary files are modified; no unrelated changes are included.
- [x] Follows clean code principles (readable, maintainable, minimal duplication).
- [x] All changes are clearly documented.
- [x] Code has been tested (manual/automated) and verified against edge cases.
- [x] No breaking changes are introduced to existing functionality.
- [x] All new and existing tests passed (if tests exist).

### Components Added:
1. **Tabs Component**: Fully accessible tab navigation with keyboard support, smooth animations, and mobile-responsive design
2. **Dropdown Menu Component**: Feature-rich dropdown with submenus, checkboxes, radio buttons, separators, and keyboard navigation
3. **Popover Component**: Floating content area with portal rendering, perfect for tooltips and contextual information

### Technical Details:
- All components use Radix UI primitives for accessibility
- Fully integrated with existing design system (Tailwind CSS utilities)
- Dark theme support out of the box
- Each component includes comprehensive props documentation
- Copy-paste ready code snippets provided

### Files Modified:
- `src/components/ui/dropdown-menu.tsx` (new)
- `src/components/ui/popover.tsx` (new)
- `src/data/components.tsx` (updated with new component entries)
- `package.json` (added new dependencies)

This enhancement significantly expands the component library and addresses the needs outlined in issue #5.

### CODE REVIEW

**Comment by fahimahammed** [comment]
please resolve conflict

File: src/data/components.tsx [hunk 1]
```diff
@@ -65,6 +65,23 @@ import {
...
+  DropdownMenuTrigger,
+  DropdownMenuCheckboxItem,
+  DropdownMenuRadioItem,
+  DropdownMenuRadioGroup,
+} from "@/components/ui/dropdown-menu";
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/components/ui/popover";
```
Comment (fahimahammed) [inline src/data/components.tsx:88]: please resolve merge conflict

**Review by fahimahammed — COMMENTED** [review]
merge conflict

**Comment by fahimahammed** [comment]
still have merge conflict

**Comment by krishkaneki** [comment]
hello @fahimahammed PR is ready for review!

**Comment by fahimahammed** [comment]
build failed

**Review by fahimahammed — APPROVED** [review]
thank you for your contribution
